lib/Widgets/HyperTextView.vala

```diff
@@ -0,0 +1,320 @@
+
+                return GLib.Source.REMOVE;
+            });
+        }
+
+        private async void update_tags_in_buffer_for_range (int buffer_start_offset, int buffer_end_offset) {
+            Gtk.TextIter buffer_start_iter, buffer_end_iter;
+            buffer.get_iter_at_offset (out buffer_start_iter, buffer_start_offset);
+            buffer_start_iter.backward_line ();
+            buffer_start_offset = buffer_start_iter.get_offset ();
+
+            buffer.get_iter_at_offset (out buffer_end_iter, buffer_end_offset);
+            buffer_end_iter.forward_line ();
+            buffer_end_offset = buffer_end_iter.get_offset ();
+
+            // Delete all tags in buffer for range [start_iter.offset,end_iter.offset]
+            lock (uri_text_tags) {
+                foreach (var tag_key in uri_text_tags.get_keys ()) {
+                    int tag_start_offset, tag_end_offset;
+                    tag_key.scanf ("[%i,%i]", out tag_start_offset, out tag_end_offset);
+
+                    if (
+                        tag_start_offset > buffer_start_offset && tag_start_offset < buffer_end_offset
+                        ||
+                        tag_end_offset > buffer_start_offset && tag_end_offset < buffer_end_offset
+                    ) {
+                        buffer.tag_table.remove (uri_text_tags.take (tag_key));
+                    }
+                }
+            }
+
+            /**
+             * Character counts are usually referred to as offsets, while byte counts are called indexes.
+             * If you confuse these two, things will work fine with ASCII, but as soon as your
+             * buffer contains multibyte characters, bad things will happen.
+             * https://developer.gnome.org/gtk3/stable/TextWidget.html
+             */
+            var buffer_start_index = buffer.text.index_of_nth_char (buffer_start_offset);
+            var buffer_end_index = buffer.text.index_of_nth_char (buffer_end_offset);
+            var buffer_substring = buffer.text.substring (buffer_start_index, buffer_end_index - buffer_start_index);
+
+            if (buffer_substring.strip () == "") {
+                // if the substring is empty, we do not have anything to do...
+                return;
+            }
+
+            // Add new tags in buffer for range [start_iter.offset,end_iter.offset]
+            GLib.MatchInfo match_info;
+            uri_regex.match (buffer_substring, 0, out match_info);
+
+            while (match_info.matches ()) {
+                string match_text = match_info.fetch (0);
+
+                /**
+                 * Character counts are usually referred to as offsets, while byte counts are called indexes.
+                 * If you confuse these two, things will work fine with ASCII, but as soon as your
+                 * buffer contains multibyte characters, bad things will happen.
+                 * https://developer.gnome.org/gtk3/stable/TextWidget.html
+                 */
```

Comment:
```suggestion
```

I think we only need one of these comments. Also I think it should be a single star comment instead of a valadoc comment, since it's not really documentation.

Reply:
Changed as suggested.